src/test/test-execute.c

```diff
@@ -764,6 +757,13 @@ int main(int argc, char *argv[]) {
         };
         int r;
 
+#ifdef __SANITIZE_ADDRESS__
+        if (is_run_on_travis_ci()) {
+                log_notice("Running on TravisCI under ASan, skipping, see https://github.com/systemd/systemd/issues/10696");
+                return 77;
```

Comment:
I've just remembered that we shouldn't be needlessly cryptic :-) I'll change it to `EXIT_TEST_SKIP` and then merge the PR after making sure that Travis CI has skipped the test as expected.